legion/legion/model/__init__.py

```diff
@@ -216,3 +216,17 @@ def define_property(name, initial_value):
         raise Exception('Context has not been defined')
 
     return _model.define_property(name, initial_value)
+
+
+def on_property_update(callback):
```

Comment:
I'd make it on_property_change as we do not distinguish b/w prop create/delete/update here.